megamek/src/megamek/server/victory/Victory.java

```diff
@@ -18,111 +18,126 @@
+	
+	public Victory(GameOptions options) {
+		checkForVictory = options.booleanOption(OptionsConstants.VICTORY_CHECK_VICTORY);
+		
+		if (checkForVictory) {
+		    VCs = buildVClist(options);
+		}
+	}
+	
+	private IVictoryConditions[] buildVClist(GameOptions options) {
+	    neededVictoryConditions = options.intOption(OptionsConstants.VICTORY_ACHIEVE_CONDITIONS);
+        ArrayList<IVictoryConditions> victories = new ArrayList<IVictoryConditions>();
+        // BV related victory conditions
+        if (options.booleanOption(OptionsConstants.VICTORY_USE_BV_DESTROYED)) {
+            victories.add(new BVDestroyedVictory(options.intOption(OptionsConstants.VICTORY_BV_DESTROYED_PERCENT)));
+        }
+        if (options.booleanOption(OptionsConstants.VICTORY_USE_BV_RATIO)) {
+            victories.add(new BVRatioVictory(options.intOption(OptionsConstants.VICTORY_BV_RATIO_PERCENT)));
+        }
+        
+        // Kill count victory condition
+        if (options.booleanOption(OptionsConstants.VICTORY_USE_KILL_COUNT)) {
+            victories.add(new KillCountVictory(options.intOption(OptionsConstants.VICTORY_GAME_KILL_COUNT)));
+        }
 
-        /**
-         * list team scores all teams which are not listed, can be assumed to be
-         * at 0.0
-         */
-        public double getTeamScore(int id);
-
-        /**
-         * list teams
-         */
-        public int[] getTeams();
-
-        /**
-         * @return true if this is a winning team id (draw == win in this case)
-         */
-        public boolean isWinningTeam(int id);
-
-        /**
-         * @return true if this is a winning player id (draw == win in this
-         *         case)
-         */
-        public boolean isWinningPlayer(int id);
+        // Commander killed victory condition
+        if (options.booleanOption(OptionsConstants.VICTORY_COMMANDER_KILLED)) {
+            victories.add(new EnemyCmdrDestroyedVictory());
+        }
+        return victories.toArray(new IVictoryConditions[0]);
+	}
+	
+    public VictoryResult checkForVictory(IGame game, HashMap<String, Object> context) {
+        VictoryResult reVal;
+        
+    	//Check for ForceVictory
+    	reVal = force.victory(game, context);
+    	if (reVal.victory()) {
+    	    return reVal;
+    	}
+    	
+    	//Check optional Victory conditions
+    	//These can have reports
+    	if (checkForVictory) {
+    	    if (VCs == null) {
+    	        VCs = buildVClist(game.getOptions());
+    	    }
+    	    reVal = checkOptionalVictory(game, context);
+    	    if (reVal.victory()) {
+    	        return reVal;
+    	    }
+    	}
+    	
+    	//Check for LastManStandingVictory
+    	VictoryResult lastManResult = lastMan.victory(game, context);
+    	if (!reVal.victory() && lastManResult.victory()) {
+    	    return lastManResult;
+    	}
+    	return reVal;
+    }
+    
+    private VictoryResult checkOptionalVictory(IGame game, HashMap<String, Object> context) {
```

Comment:
Should be `Map<String, Object> context` instead of `HashMap<String, Object> context`.